piet-web/src/lib.rs

```diff
@@ -2,20 +2,25 @@
 
 use std::borrow::Cow;
 
-use wasm_bindgen::JsValue;
-use web_sys::{CanvasRenderingContext2d, CanvasWindingRule};
+use wasm_bindgen::{Clamped, JsCast, JsValue};
+use web_sys::{CanvasRenderingContext2d, CanvasWindingRule, HtmlCanvasElement, ImageData, Window};
 
-use kurbo::{Affine, PathEl, Shape, Vec2};
+use kurbo::{Affine, PathEl, Rect, Shape, Vec2};
 
-use piet::{Font, FontBuilder, RenderContext, RoundInto, TextLayout, TextLayoutBuilder};
+use piet::{
+    Font, FontBuilder, ImageFormat, InterpolationMode, RenderContext, RoundInto, TextLayout,
+    TextLayoutBuilder,
+};
 
 pub struct WebRenderContext<'a> {
     ctx: &'a mut CanvasRenderingContext2d,
+    /// Used for creating image bitmaps and possibly other resources.
+    window: &'a Window,
```

Comment:
My gut reaction to this was "wow that's taking a large dependency". I think this is something to keep an eye on for the future to see if there's a better way

Reply:
100%. I'm willing to consider making this a feature, and having some image operations fail. I also feel that this is one of many weaknesses in the web canvas spec - it can't decide whether it's a lightweight JS API (its origins from Apple) or deeply bound into the whole web stack (certainly the new ImageBitmap mechanism).